README.md

```diff
@@ -99,38 +96,146 @@ Running `flask db migrate` on the container ci_app_1 errors because the files in
 
 Build the "ci_test" Docker image by running this command:
 
-```
+```bash
 docker-compose -f ci/docker-compose-test.yml build test
 ```
 
 **Rebuild ci_test whenever Dockerfile.test, requirements_for_test.txt, or the notification_api image changes.**
 
 To run all unit tests:
 
-```
+```bash
 docker-compose -f ci/docker-compose-test.yml up --abort-on-container-exit
 ```
 
 The Github workflow also runs these tests when you push code.  Instructions for running a subset of tests are located in tests/README.md.
 
+### Pre-commit hooks
+
+This repository uses [pre-commit](https://pre-commit.com/) and [talisman](https://github.com/thoughtworks/talisman) to scan changes for keys and secrets.  To set it up, install the required dependencies `pre-commit` and `go`.
+
+OSX users can run `brew bundle` and then `pre-commit install` to register the git hooks.  The configuration is stored in .pre-commit-config.yaml.
+
 ### Building the production application container
 
 To verify that the production application container build should succeed during deployment, run:
 
-```
+```bash
 docker-compose -f ci/docker-compose.yml up --build --abort-on-container-exit
 ```
 
 Note that the production infrastructure does not use docker-compose.yml.
 
-###
-
-Using Localstack
+### Using Localstack
 
 TODO
 
 ---
 
+
+## Local Development without docker
+
+### Prerequisite installation
+
+ On OS X:
+
+ 1. Install PyEnv with Homebrew. This will preserve your sanity.
+
+  `brew install pyenv`
+
+ 2. Install Python 3.6.10 (or whatever version is specified in .python-version)
+ Then follow from instructions for rest of pyenv setup, [see step 3 here](https://github.com/pyenv/pyenv#basic-github-checkout)
+
+ Note: For MacOS devs who are using Big Sur, Monterey, standard pyenv python installation will be failed in most case. I found [this solution](https://github.com/pyenv/pyenv/issues/2143#issuecomment-1070640288) so only 3.7.13, 3.8.13, 3.9.11 and 3.10.3 works fine.
+
+ `pyenv install 3.6.10`
+
+ 3. If you expect no conflicts, set `3.6.10` as you default
```

Comment:
Should these be `3.8` instead of `3.6`?

Reply:
Yes. Or a more specific version of 3.8.  3.8.14 was just released earlier this month, and 3.8 itself was released in 2019. Have we standardized on any specific version?

Reply:
FYI on my Mac M1 (silicon), I'm using 3.8.13, which is what that note says on line 149, so I think that should be the version.